jupyter_scheduler/executors.py

```diff
@@ -221,3 +222,66 @@ def execute(self):
             archive_filepath = self.staging_paths["tar.gz"]
             with fsspec.open(archive_filepath, "wb") as f:
                 f.write(fh.getvalue())
+
+
+class AllFilesArchivingExecutionManager(DefaultExecutionManager):
+    """Execution manager that archives all output files in and under the
+    output directory into a single archive file
+
+    Notes
+    -----
+    Should be used along with :class:`~jupyter_scheduler.scheduler.AllFilesArchivingScheduler`
+    as the `scheduler_class` during jupyter server start.
+    """
+
+    def execute(self):
+        job = self.model
+
+        with open(self.staging_paths["input"], encoding="utf-8") as f:
+            nb = nbformat.read(f, as_version=4)
+
+        if job.parameters:
+            nb = add_parameters(nb, job.parameters)
+
+        ep = ExecutePreprocessor(
+            kernel_name=nb.metadata.kernelspec["name"],
+            store_widget_state=True,
+        )
+
+        # Get the directory of the input file
+        working_dir = os.path.dirname(os.path.abspath(self.staging_paths["input"]))
```

Comment:
There's no need to call `os.path.abspath()` here, since the staging paths are all absolute paths.